modules/itc/src/main/scala/lucuma/itc/Itc.scala

```diff
@@ -90,6 +92,63 @@ object Itc:
     charts:      NonEmptyList[ItcChartGroup]
   )
 
+  object GraphResult:
+    def fromLegacy(
+      dataVersion: String,
+      ccds:        NonEmptyList[ItcRemoteCcd],
+      charts:      NonEmptyList[ItcChartGroup]
+    ): GraphResult = {
+      println(
+        charts
+          .flatMap(_.charts)
+          .filter(_.chartType === ChartType.S2NChart)
+          .map(_.series.filter(_.seriesType === SeriesDataType.FinalS2NData))
+          .map(_.map(_.title))
+      )
+
+      def maxWavelength(chart: ItcChart, seriesDataType: SeriesDataType): List[Wavelength] =
+        chart.series
+          .filter(_.seriesType === seriesDataType)
+          .zip(ccds.toList)
+          .map { case (sn, ccd) =>
+            sn.data
+              .maxByOption(_._2)
+              .map(_._1)
+              .flatMap(i => Wavelength.fromPicometers.getOption((i * 1000).toInt))
+          }
+          .flattenOption
+      // Calculate the wavelengths at where the peaks happen
+      val calculatedCCDs                                                                   =
+        charts
+          .flatMap(_.charts)
+          .filter(_.chartType === ChartType.S2NChart)
+          .flatMap { chart =>
+            val maxFinalAt  = maxWavelength(chart, SeriesDataType.FinalS2NData)
+            val maxSignalAt = maxWavelength(chart, SeriesDataType.SingleS2NData)
+
+            ccds.zipWithIndex
+              .map { (ccd, i) =>
+                val finalWV  = maxFinalAt.lift(i)
+                val singleWV = maxFinalAt.lift(i)
+                (finalWV, singleWV).mapN { (maxFinalAt, maxSingleAt) =>
+                  ItcCcd(ccd.singleSNRatio,
+                         ccd.totalSNRatio,
+                         maxFinalAt,
+                         maxSingleAt,
+                         ccd.peakPixelFlux,
+                         ccd.wellDepth,
+                         ccd.ampGain,
+                         ccd.warnings
+                  )
+                }
+              }
+              .toList
+              .flattenOption
+          }
+      assert(calculatedCCDs.length === ccds.length)
```

Comment:
Is the assert meant to be left in final code?

Reply:
Yes I think so, this may alert us to logical errors if they ever happen